app/blog/create/page.tsx

```diff
@@ -111,36 +128,88 @@ function CreatePost() {
         if (uploadError) {
           throw uploadError;
         }
-        console.log(uploadError)
+        console.log(uploadError);
       }
-      // attempt to add post to DB, throws error
-      const { error: postInsertError } = await supabase.from("posts").insert({
-        entry,
-        author: user?.id,
-        title: postTitle,
-        first_paragraph: firstParagraph,
-        image: imagePath, // will get this after uploading image
-      });
+      // attempt to insert or upsert post to DB, throws error
+      if (id) {
+        await updatePost(imagePath);
+      } else {
+        await initialPost(imagePath);
+      }
+
+      setSaveStatus("Draft saved.");
 
-      if (postInsertError) {
-        throw postInsertError;
+      if (publish) {
+        router.refresh();
+        router.push("/blog");
       }
-      router.refresh()
-      router.push("/blog");
     } catch (error: any) {
       setPostCreationError(error);
       setSubmitting(false);
+      setSaveStatus(null);
+    }
+  }
+
+  async function initialPost(imagePath: string | null) {
+    const entry = tipTapEditor?.getJSON();
+    const firstParagraph = tipTapEditor?.getText() || "";
+    const { data, error: postInsertError } = await supabase
+      .from("posts")
+      .insert({
+        entry,
+        author: user?.id,
+        title: postTitleRef.current,
+        first_paragraph:
+          firstParagraph?.length > 300
+            ? firstParagraph?.substring(0, 300) + "..."
+            : firstParagraph,
+        image: imagePath, // will get this after uploading image
+        published: false,
+      })
+      .select();
+
+    if (postInsertError) {
+      console.log("Error inserting post: ", postInsertError);
```

Comment:
same here.. lets add some error handling later on for this but clear this out for now